server/src/main/java/org/candlepin/jackson/CandlepinAttributesDeserializer.java

```diff
@@ -0,0 +1,178 @@
+/**
+ * Copyright (c) 2009 - 2016 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.jackson;
+
+import org.candlepin.common.exceptions.CandlepinJsonProcessingException;
+import org.candlepin.dto.api.v1.AttributesDTO;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.TreeNode;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+
+/**
+ * The CandlepinAttributeDeserializer handles the deserialization of attributes, processing both
+ * the current attribute mapping style and obsoleted formats.
+ */
+public class CandlepinAttributesDeserializer extends StdDeserializer<AttributesDTO> {
```

Comment:
Does this have value when we're explicitly defining the structure via objects (and inheriting oddities such as the ```.getAttributes().getAttributes()``` as above?)

Reply:
Done. Not needed anymore after removal of ProductLegacyDTO.